### DISCUSSION TITLE
Add icon in the input after validation passed or not

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#8373] [Help] [answered]
How can I check in the filament form text inputs if the validation is passed, and then show such an icon?
![image](https://github.com/filamentphp/filament/assets/29732308/5cf4e313-cf52-4707-86fc-62d8316fdd88)

**Comment by leandrocfe** [comment, marked as answer]
A workaround
```php
use Filament\Forms\Components\Component;
use Livewire\Component as Livewire;

TextInput::make('field')
    ->required()
    ->minLength(3)
    ->live(onBlur: true)
    ->afterStateUpdated(function (Livewire $livewire, Component $component) {
        $validate = $livewire->validateOnly($component->getStatePath());
        $component
            ->hintIcon('heroicon-m-check-circle')
            ->hintColor('success');
    })
```
You can also use `suffixIcon`
I think it might be slow if you have a lot of fields

**Reply by ArtMin96** [reply]
Thank you. This is what I needed. How can I change the color of `suffixIcon`?

**Reply by leandrocfe** [reply]
Hum, I think you can't

**Reply by ArtMin96** [reply]
Ok. Thanks again
